app/models/submission.rb

```diff
@@ -482,6 +498,268 @@ def turnitinable_by_lti?
     turnitin_data.select{|_, v| v.is_a?(Hash) && v.key?(:outcome_response)}.any?
   end
 
+  # VeriCite
+
+  # this function will check if the score needs to be updated and update/save the new score if so,
+  # otherwise, it just returns the vericite_data_hash
+  def vericite_data()
+    self.vericite_data_hash ||= {}
+    self.vericite_data_hash[:provider] = :vericite
+    # check to see if the score is stale, if so, fetch it again
+    updateScores = false
+    # since there could be multiple versions, let's not waste calls for old versions and use the old score
+    if !self.readonly? && self.versions.current.number == self.version_number
+      self.vericite_data_hash.keys.each do |asset_string|
+        data = self.vericite_data_hash[asset_string]
+        next unless data && data.is_a?(Hash) && data[:object_id]
+        updateScores = updateScores || vericite_recheck_score(data)
+      end
+      # we have found at least one score that is stale, call VeriCite and save the results
+      if updateScores
+        check_vericite_status
+      end
```

Comment:
I just remembered that the "self.readonly" variable is set to "true" for the gradebook main page (i.e. the instructor view where all students and all assignments are listed). This means that this page will never consult VeriCite's API for the scores and only show what is stored in the Canvas DB (which it already has in the object). This is also the reason why I cache the scores based on SiteId + AssignmentId + UserID, since there is no other screen where VeriCite scores show up where it isn't just one user. Sorry about not remembering this until now, I wrote this code back in April and its hard to remember these small nuances. You can easily validate this by adding a log message in the beginning of this function:

Rails.logger.info "Readonly: #{self.readonly?}"